src/etl/sds/worker/transform_bulk/tests/test_transform_bulk_worker.py

```diff
@@ -134,38 +142,44 @@ def test_transform_worker_pass_no_dupes_max_records(
     put_object(key=WorkerKey.TRANSFORM, body=pkl_dumps_lz4(initial_unprocessed_data))
     put_object(key=WorkerKey.LOAD, body=pkl_dumps_lz4(initial_processed_data))
 
-    n_total_processed_records_expected = 0
-    n_unprocessed_records = n_initial_unprocessed
-    while n_unprocessed_records > 0:
-        n_newly_processed_records_expected = min(n_unprocessed_records, max_records)
-        n_unprocessed_records_expected = (
-            n_unprocessed_records - n_newly_processed_records_expected
-        )
-        n_total_processed_records_expected += n_newly_processed_records_expected
+    expected_responses = []
+    expected_iterations = min(
+        ceil(n_initial_unprocessed / max_records), transform_bulk.FAN_OUT
```

Comment:
Is rounding this value up ever going to get it wrong?

Reply:
the rounding is deliberate, e.g. with `transform_bulk.FAN_OUT = 10` (which the hard-coded value) then the behaviour of that line is if there are 1007 initial records and each "fan out" processes `max_records=100` then  `ceil(1007 / 100)=11` meaning that we would need to do 11 "fan out" processes (so there are 10 "fan outs" of 100 records and an 11th of the remaining 7 records). The test shows that it will only process the first 1000 in this case, with 7 remaining records to be processed in the next iteration of the etl

Reply:
Ahhh gotcha!